### PR TITLE
Update docs for develop

### PR DESCRIPTION
**Pull request by wzshiming**
#### What type of PR is this?

/kind documentation

#### Which issue(s) this PR fixes:

xref #269

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
NONE
```

### CODE REVIEW

File: site/content/en/docs/user/without-container.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+```
+
+The binaries will be located in `~/.kwok/cache/kubernetes/${KUBE_VERSION}/_output/local/bin/$(go env GOOS)/$(go env GOARCH)`.
+
+Now, we can create cluster using `kwokctl` with `binary` runtime.
+
+``` bash
+kwokctl create cluster \
+  --runtime=binary
+```
```
Comment (Zhuzhenghao) [inline site/content/en/docs/user/without-container.md:42]: After I build on my MacOS, I run the command `kwokctl create cluster --runtime=binary`, errors ars as follow:

```shell
kwokctl create cluster \
  --runtime=binary
Cluster is creating                                                                               cluster=kwok
Download uri=https://github.com/kubernetes-sigs/kwok/releases/download/v0.2.0-96-gc4ee59d/kwok-darwin-arm64 cluster=kwok
ERROR Failed to setup config err="https://github.com/kubernetes-sigs/kwok/releases/download/v0.2.0-96-gc4ee59d/kwok-darwin-arm64: 404 Not Found" cluster=kwok
```

Comment (wzshiming) [reply site/content/en/docs/user/without-container.md:42]: the `v0.2.0-96-gc4ee59d` only exist in your local, you should specify the kwok-controller-binary or use the latest version

https://github.com/kubernetes-sigs/kwok/releases/tag/v0.2.0

File: site/content/en/docs/contributing/development.md [hunk 2]
```diff
@@ -13,32 +13,53 @@ This document provides details on how to build and run `kwok` and `kwokctl` loca
...
+Build `kwok` and `kwokctl` binaries.
+
+``` bash
+make build
 ```
 
-The image will be tagged as `local/kwok:${tag}` and can be found in `docker images`.
+On a successful build, the binaries will be located in `./bin/$(go env GOOS)/$(go env GOARCH)`.
+
+Note that if running in Non-Linux then you will need take look [Without Container] to build the Kubernetes binaries.
```
Comment (Huang-Wei) [inline site/content/en/docs/contributing/development.md:55]:
```suggestion
Note that if running in Non-Linux platforms, then you will need to follow [Without Container] to build the Kubernetes binaries locally.
```

File: site/content/en/_index.md [hunk 3]
```diff
@@ -63,7 +63,7 @@ Runtime indicates which medium `kwokctl` will use to start the cluster
 | Windows | 🟠/🔵    | 🔵      | 🔵      | 🔵       | 🔵    |
 
 - 🟢 Supported
-- 🟠 Need to use your own build of the Kubernetes binary
+- 🟠 Need to use your own build of the [Kubernetes binary without Container]
```
Comment (Huang-Wei) [inline site/content/en/_index.md:66]: the term "without container" doesn't sound that conventional, how about:

```suggestion
- 🟠 Need to build [platform-specific Kubernetes binaries]
```

And we replace "without container" with "platform-specific Kubernetes binaries"

File: site/content/en/docs/contributing/development.md [hunk 4]
```diff
@@ -13,32 +13,53 @@ This document provides details on how to build and run `kwok` and `kwokctl` loca
...
-IMAGE_PREFIX=local make build
+./bin/$(go env GOOS)/$(go env GOARCH)/kwokctl create cluster \
+  --runtime=docker
 ```
 
-On a successful build, the binaries will be located in `./bin/$(go env GOOS)/$(go env GOARCH)`.
+By the way, you can also use `podman` or `nerdctl` as the builder and the runtime.
 
-### Building `kwok` image
+### Start without Container
```
Comment (Huang-Wei) [inline site/content/en/docs/contributing/development.md:45]:
```suggestion
### Start with platform-specific binaries
```

File: site/content/en/docs/contributing/development.md [hunk 5]
```diff
@@ -13,32 +13,53 @@ This document provides details on how to build and run `kwok` and `kwokctl` loca
...
+    - v1alpha1 - API definitions for parsing and converting only
   - config - Configuration utilities
   - kwok - `kwok` implementation
   - kwokctl - `kwokctl` implementation
 
-## Building
+## Build and Run
 
-### Building `kwok` and `kwokctl`
+### Start with Container
```
Comment (Huang-Wei) [inline site/content/en/docs/contributing/development.md:26]:
```suggestion
### Start with Containers
```

File: site/content/en/docs/user/without-container.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
+# Without Container
```
Comment (Huang-Wei) [inline site/content/en/docs/user/without-container.md:1]: ditto, maybe rename the title as well as filename.

**Comment by wzshiming** [comment]
updated

**Pull request merged**